src/client/components/forms/publisher.js

```diff
@@ -29,7 +30,18 @@ const LoadingSpinner = require('../loading-spinner');
 const PublisherData = require('./parts/publisher-data');
 const RevisionNote = require('./parts/revision-note');
 
+/**
+ * React component to define a form to create and edit
+ * a Publisher.
+ * @extends React.Component
+ */
+
 class PublisherForm extends React.Component {
+
+	/**
+	 * Binds class methods to their respective data.
```

Comment:
I'm not too keen on this description. I think this would be better:

	/**
	 * Constructs a new instance of the PublisherForm class.
	 * This constructor calls the constructor of the React.Component class,
	 * initializes the component state and binds class methods to the class
	 * instance.
	 * @constructor
	 *
	 * @props {object} props - The properties of the component
	 **/